### PR TITLE
Used assertRaisesRegexp instead of assertRaises

### LINKED ISSUE #781: Use assertRaisesRegexp instead of assertRaises
## Description
`assertRaises` is always lazy and should be avoided, instead we should use `assertRaisesRegexp` which provides an extra field for regex match with exception message.
There are a total of 49 uses of assertRaises in 10 files as per and can be checked using `git grep`

## Mocks
ss

https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaisesRegex

### Update [Required]
- [ ] Modify existing use of assertRaises to assertRaisesRegexp

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 2 mentor.

## Estimation
2-3 hours

### PR DESCRIPTION
**Pull request by sanjana11147**
### **Description**
Modified the existing use of assertRaises to assertRaisesRegexp which provides an extra field for regex match with the exception message.

This is my first PR. Let me know if you have any feedback!

Fixes #781 

# Type of Change:

- Code
- Quality Assurance

**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?
Locally

# Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] Any dependent changes have been merged 

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings 
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been published in downstream modules

### CODE REVIEW

**Comment by Monal5031** [comment]
@sanjana11147 travis fails

**Comment by sanjana11147** [comment]
Can you help me here @Monal5031 
Even though the regex message is the same as it should be, it's still failing in two instances.

**Comment by sanjana11147** [comment]
What's the problem with the changes I made? @Monal5031

File: aut_docs/Tutorial.md [hunk 1]
```diff
@@ -40,7 +40,7 @@ Lets start by importing this TestCase and a few other necessary imports.
             @classmethod
             def setUpClass(cls):
                 cls.driver = webdriver.Firefox()
-                cls.driver.implicitly_wait(5)
+                cls.driver.implicitly_wait(25)
```
Comment (Monal5031) [inline aut_docs/Tutorial.md]: No... never change the values in `implicitly_wait`

File: aut_docs/Tutorial.md [hunk 2]
```diff
@@ -162,7 +162,7 @@ After compiling all the above code snippets the test finally will look like:
         @classmethod
         def setUpClass(cls):
             cls.driver = webdriver.Firefox()
-            cls.driver.implicitly_wait(5)
+            cls.driver.implicitly_wait(25)
```
Comment (Monal5031) [inline aut_docs/Tutorial.md]: Same here

File: vms/administrator/tests/test_formFields.py [hunk 3]
```diff
@@ -35,7 +35,7 @@ def setUpClass(cls):
         firefox_options = Options()
         firefox_options.add_argument('-headless')
         cls.driver = webdriver.Firefox(firefox_options=firefox_options)
-        cls.driver.implicitly_wait(5)
+        cls.driver.implicitly_wait(25)
```
Comment (Monal5031) [inline vms/administrator/tests/test_formFields.py]: Same here

File: vms/administrator/tests/test_report.py [hunk 4]
```diff
@@ -32,7 +32,7 @@ def setUpClass(cls):
         firefox_options = Options()
         firefox_options.add_argument('-headless')
         cls.driver = webdriver.Firefox(firefox_options=firefox_options)
-        cls.driver.implicitly_wait(5)
+        cls.driver.implicitly_wait(25)
```
Comment (Monal5031) [inline vms/administrator/tests/test_report.py]: Same here

File: vms/administrator/tests/test_settings.py [hunk 5]
```diff
@@ -72,7 +72,7 @@ def setUpClass(cls):
         firefox_options = Options()
         firefox_options.add_argument('-headless')
         cls.driver = webdriver.Firefox(firefox_options=firefox_options)
-        cls.driver.implicitly_wait(5)
+        cls.driver.implicitly_wait(50)
```
Comment (Monal5031) [inline vms/administrator/tests/test_settings.py]: Same here

File: vms/event/tests/test_event.py [hunk 6]
```diff
@@ -31,7 +31,7 @@ def setUpClass(cls):
         firefox_options = Options()
         firefox_options.add_argument('-headless')
         cls.driver = webdriver.Firefox(firefox_options=firefox_options)
-        cls.driver.implicitly_wait(5)
+        cls.driver.implicitly_wait(25)
```
Comment (Monal5031) [inline vms/event/tests/test_event.py]: Same here

File: vms/event/tests/test_searchEvent.py [hunk 7]
```diff
@@ -46,7 +46,7 @@ def setUpClass(cls):
         firefox_options = Options()
         firefox_options.add_argument('-headless')
         cls.driver = webdriver.Firefox(firefox_options=firefox_options)
-        cls.driver.implicitly_wait(5)
+        cls.driver.implicitly_wait(25)
```
Comment (Monal5031) [inline vms/event/tests/test_searchEvent.py]: Same here

File: vms/event/tests/test_shiftSignUp.py [hunk 8]
```diff
@@ -40,7 +40,7 @@ def setUpClass(cls):
         firefox_options = Options()
         firefox_options.add_argument('-headless')
         cls.driver = webdriver.Firefox(firefox_options=firefox_options)
-        cls.driver.implicitly_wait(5)
+        cls.driver.implicitly_wait(50)
```
Comment (Monal5031) [inline vms/event/tests/test_shiftSignUp.py]: Same here

File: vms/event/tests/test_shiftSignUp.py [hunk 9]
```diff
@@ -141,9 +142,10 @@ def test_signup_for_same_shift_again(self):
         sign_up_page.navigate_to_sign_up()
 
         # Events shown in table
-        with self.assertRaises(NoSuchElementException):
-            sign_up_page.get_info_box()
-        sign_up_page.click_to_view_jobs()
+        with self.assertRaisesRegexp(NoSuchElementException,
```
Comment (Monal5031) [inline vms/event/tests/test_shiftSignUp.py]: no use of `with` when using `assertRaisesRegexp`

File: vms/event/tests/test_shiftSignUp.py [hunk 10]
```diff
@@ -184,9 +188,10 @@ def test_empty_events(self):
             sign_up_page.no_event_message
         )
 
-        with self.assertRaises(NoSuchElementException):
-            sign_up_page.find_table_tag()
-            sign_up_page.click_to_view_jobs()
+        with self.assertRaisesRegexp(NoSuchElementException,
```
Comment (Monal5031) [inline vms/event/tests/test_shiftSignUp.py]: no use of `with` when using `assertRaisesRegexp`

File: vms/home/tests/test_functional.py [hunk 11]
```diff
@@ -35,7 +35,7 @@ def setUpClass(cls):
         firefox_options = Options()
         firefox_options.add_argument('-headless')
         cls.driver = webdriver.Firefox(firefox_options=firefox_options)
-        cls.driver.implicitly_wait(5)
+        cls.driver.implicitly_wait(25)
```
Comment (Monal5031) [inline vms/home/tests/test_functional.py]: Same here

File: vms/home/tests/test_functional.py [hunk 12]
```diff
@@ -199,7 +199,7 @@ def setUpClass(cls):
...
```
Comment (Monal5031) [inline vms/home/tests/test_functional.py]: Same here

File: vms/job/tests/test_jobDetails.py [hunk 13]
```diff
@@ -32,7 +32,7 @@ def setUpClass(cls):
         firefox_options = Options()
         firefox_options.add_argument('-headless')
         cls.driver = webdriver.Firefox(firefox_options=firefox_options)
-        cls.driver.implicitly_wait(5)
+        cls.driver.implicitly_wait(25)
```
Comment (Monal5031) [inline vms/job/tests/test_jobDetails.py]: Same here

File: vms/job/tests/test_searchJob.py [hunk 14]
```diff
@@ -72,7 +72,7 @@ def setUpClass(cls):
         firefox_options = Options()
         firefox_options.add_argument('-headless')
         cls.driver = webdriver.Firefox(firefox_options=firefox_options)
-        cls.driver.implicitly_wait(5)
+        cls.driver.implicitly_wait(25)
```
Comment (Monal5031) [inline vms/job/tests/test_searchJob.py]: Same here

File: vms/organization/tests/test_organization.py [hunk 15]
```diff
@@ -37,7 +37,7 @@ def setUpClass(cls):
         firefox_options = Options()
         firefox_options.add_argument('-headless')
         cls.driver = webdriver.Firefox(firefox_options=firefox_options)
-        cls.driver.implicitly_wait(5)
+        cls.driver.implicitly_wait(35)
```
Comment (Monal5031) [inline vms/organization/tests/test_organization.py]: Same here

File: vms/registration/tests/test_functional_admin.py [hunk 16]
```diff
@@ -67,7 +67,7 @@ def setUpClass(cls):
         firefox_options = Options()
         firefox_options.add_argument('-headless')
         cls.driver = webdriver.Firefox(firefox_options=firefox_options)
-        cls.driver.implicitly_wait(5)
+        cls.driver.implicitly_wait(25)
```
Comment (Monal5031) [inline vms/registration/tests/test_functional_admin.py]: Same here

File: vms/registration/tests/test_functional_volunteer.py [hunk 17]
```diff
@@ -66,7 +66,7 @@ def setUpClass(cls):
         firefox_options = Options()
         firefox_options.add_argument('-headless')
         cls.driver = webdriver.Firefox(firefox_options=firefox_options)
-        cls.driver.implicitly_wait(5)
+        cls.driver.implicitly_wait(25)
```
Comment (Monal5031) [inline vms/registration/tests/test_functional_volunteer.py]: Same here

File: vms/shift/tests/test_manageVolunteerShift.py [hunk 18]
```diff
@@ -62,7 +62,7 @@ def setUpClass(cls):
         firefox_options = Options()
         firefox_options.add_argument('-headless')
         cls.driver = webdriver.Firefox(firefox_options=firefox_options)
-        cls.driver.implicitly_wait(5)
+        cls.driver.implicitly_wait(25)
```
Comment (Monal5031) [inline vms/shift/tests/test_manageVolunteerShift.py]: Same here

File: vms/shift/tests/test_services.py [hunk 19]
```diff
@@ -456,13 +456,13 @@ def test_cancel_shift_registration(self):
         """ Uses shifts s1, s2, s3 and volunteers v1,v2 """
 
         # test cases when try to cancel when they aren't signed up for a shift
-        with self.assertRaises(ObjectDoesNotExist):
+        with self.assertRaisesRegexp(ObjectDoesNotExist, ''):
```
Comment (Monal5031) [inline vms/shift/tests/test_services.py]: No using `with` when using `assertRaisesRegexp`

File: vms/shift/tests/test_services.py [hunk 20]
```diff
@@ -456,13 +456,13 @@ def test_cancel_shift_registration(self):
...
             cancel_shift_registration(self.v1.id, self.s1.id)
 
-        with self.assertRaises(ObjectDoesNotExist):
+        with self.assertRaisesRegexp(ObjectDoesNotExist, ''):
```
Comment (Monal5031) [inline vms/shift/tests/test_services.py]: No using `with` when using `assertRaisesRegexp`

File: vms/shift/tests/test_services.py [hunk 21]
```diff
@@ -456,13 +456,13 @@ def test_cancel_shift_registration(self):
...
             cancel_shift_registration(self.v1.id, self.s2.id)
 
-        with self.assertRaises(ObjectDoesNotExist):
+        with self.assertRaisesRegexp(ObjectDoesNotExist, ''):
```
Comment (Monal5031) [inline vms/shift/tests/test_services.py]: No using `with` when using `assertRaisesRegexp`

File: vms/shift/tests/test_services.py [hunk 22]
```diff
@@ -473,7 +473,7 @@ def test_cancel_shift_registration(self):
         # test typical cases
         cancel_shift_registration(self.v1.id, self.s1.id)
 
-        with self.assertRaises(ObjectDoesNotExist):
+        with self.assertRaisesRegexp(ObjectDoesNotExist, ''):
```
Comment (Monal5031) [inline vms/shift/tests/test_services.py]: No using `with` when using `assertRaisesRegexp`

File: vms/shift/tests/test_shiftDetails.py [hunk 23]
```diff
@@ -37,7 +37,7 @@ def setUpClass(cls):
         firefox_options = Options()
         firefox_options.add_argument('-headless')
         cls.driver = webdriver.Firefox(firefox_options=firefox_options)
-        cls.driver.implicitly_wait(5)
+        cls.driver.implicitly_wait(25)
```
Comment (Monal5031) [inline vms/shift/tests/test_shiftDetails.py]: Same here

File: vms/shift/tests/test_shiftHours.py [hunk 24]
```diff
@@ -36,7 +36,7 @@ def setUpClass(cls):
         firefox_options = Options()
         firefox_options.add_argument('-headless')
         cls.driver = webdriver.Firefox(firefox_options=firefox_options)
-        cls.driver.implicitly_wait(5)
+        cls.driver.implicitly_wait(25)
```
Comment (Monal5031) [inline vms/shift/tests/test_shiftHours.py]: Same here

File: vms/shift/tests/test_viewVolunteerShift.py [hunk 25]
```diff
@@ -49,7 +49,7 @@ def setUpClass(cls):
         firefox_options = Options()
         firefox_options.add_argument('-headless')
         cls.driver = webdriver.Firefox(firefox_options=firefox_options)
-        cls.driver.implicitly_wait(5)
+        cls.driver.implicitly_wait(25)
```
Comment (Monal5031) [inline vms/shift/tests/test_viewVolunteerShift.py]: Same here

File: vms/volunteer/tests/test_searchVolunteer.py [hunk 26]
```diff
@@ -56,7 +56,7 @@ def setUpClass(cls):
         firefox_options = Options()
         firefox_options.add_argument('-headless')
         cls.driver = webdriver.Firefox(firefox_options=firefox_options)
-        cls.driver.implicitly_wait(5)
+        cls.driver.implicitly_wait(25)
```
Comment (Monal5031) [inline vms/volunteer/tests/test_searchVolunteer.py]: Same here

File: vms/volunteer/tests/test_volunteerProfile.py [hunk 27]
```diff
@@ -44,7 +44,7 @@ def setUpClass(cls):
         firefox_options = Options()
         firefox_options.add_argument('-headless')
         cls.driver = webdriver.Firefox(firefox_options=firefox_options)
-        cls.driver.implicitly_wait(5)
+        cls.driver.implicitly_wait(25)
```
Comment (Monal5031) [inline vms/volunteer/tests/test_volunteerProfile.py]: Same here

File: vms/volunteer/tests/test_volunteerReport.py [hunk 28]
```diff
@@ -33,7 +33,7 @@ def setUpClass(cls):
         firefox_options = Options()
         firefox_options.add_argument('-headless')
         cls.driver = webdriver.Firefox(firefox_options=firefox_options)
-        cls.driver.implicitly_wait(5)
+        cls.driver.implicitly_wait(25)
```
Comment (Monal5031) [inline vms/volunteer/tests/test_volunteerReport.py]: Same here

**Comment by Monal5031** [comment]
Closing due to inactivity

**Pull request closed without merging**